clientv3/integration/fragment_test.go

```diff
@@ -0,0 +1,161 @@
+)
+
+func TestResponseWithoutFragmenting(t *testing.T) {
+	defer testutil.AfterTest(t)
+	// MaxResponseBytes will overflow to 1000 once the grpcOverheadBytes,
+	// which have a value of 512 * 1024, are added to MaxResponseBytes.
+	clus := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 3, MaxResponseBytes: ^uint(0) - (512*1024 - 1 - 1000)})
+	defer clus.Terminate(t)
+
+	cfg := clientv3.Config{
+		Endpoints: []string{
+			clus.Members[0].GRPCAddr(),
+			clus.Members[1].GRPCAddr(),
+			clus.Members[2].GRPCAddr(),
+		},
+	}
+	cli, err := clientv3.New(cfg)
+
+	cli.SetEndpoints(clus.Members[0].GRPCAddr())
+	firstLease, err := clus.Client(0).Grant(context.Background(), 10000)
+	if err != nil {
+		t.Error(err)
+	}
+
+	// Create and register watch proxy.
+	wp, _ := grpcproxy.NewWatchProxy(clus.Client(0))
+	wc := adapter.WatchServerToWatchClient(wp)
+	w := clientv3.NewWatchFromWatchClient(wc)
+
+	kv := clus.Client(0)
+	for i := 0; i < 10; i++ {
+		_, err = kv.Put(context.TODO(), fmt.Sprintf("foo%d", i), "bar", clientv3.WithLease(firstLease.ID))
+		if err != nil {
+			t.Error(err)
+		}
+	}
+
+	// Does not include the clientv3.WithFragmentedResponse option.
+	wChannel := w.Watch(context.TODO(), "foo", clientv3.WithRange("z"))
+	_, err = clus.Client(0).Revoke(context.Background(), firstLease.ID)
+	if err != nil {
+		t.Error(err)
+	}
+
+	r, ok := <-wChannel
+	if !ok {
+		t.Error()
+	}
+	keyDigitSum := 0
+	responseSum := 0
+	if len(r.Events) != 10 {
+		t.Errorf("Expected 10 events, got %d\n", len(r.Events))
+	}
+	for i := 0; i < 10; i++ {
+		if r.Events[i].Type != mvccpb.DELETE {
+			t.Errorf("Expected DELETE event, got %d", r.Events[i].Type)
+		}
+		keyDigitSum += i
+		digit, err := strconv.Atoi((string(r.Events[i].Kv.Key)[3:]))
+		if err != nil {
+			t.Error("Failed to convert %s to int", (string(r.Events[i].Kv.Key)[3:]))
+		}
+		responseSum += digit
+	}
+	if keyDigitSum != responseSum {
+		t.Errorf("Expected digits of keys received in the response to sum to %d, but got %d\n", keyDigitSum, responseSum)
+	}
+}
+
+func TestFragmenting(t *testing.T) {
+	defer testutil.AfterTest(t)
+	// MaxResponseBytes will overflow to 1000 once the grpcOverheadBytes,
+	// which have a value of 512 * 1024, are added to MaxResponseBytes.
+	clus := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 3, MaxResponseBytes: ^uint(0) - (512*1024 - 1 - 1000)})
+	defer clus.Terminate(t)
+
+	cfg := clientv3.Config{
+		Endpoints: []string{
+			clus.Members[0].GRPCAddr(),
+			clus.Members[1].GRPCAddr(),
+			clus.Members[2].GRPCAddr(),
+		},
+	}
+	cli, err := clientv3.New(cfg)
+
+	cli.SetEndpoints(clus.Members[0].GRPCAddr())
+	firstLease, err := clus.Client(0).Grant(context.Background(), 10000)
+	if err != nil {
+		t.Error(err)
+	}
+
+	// Create and register watch proxy
+	wp, _ := grpcproxy.NewWatchProxy(clus.Client(0))
+	wc := adapter.WatchServerToWatchClient(wp)
+	w := clientv3.NewWatchFromWatchClient(wc)
+
+	kv := clus.Client(0)
+	for i := 0; i < 100; i++ {
```

Comment:
probably do this concurrently; 100 puts is a little pricey

Reply:
Where is the flag to concurrently put?